File: ros2_fundamentals/start_with_simple_nodes_py/start_with_simple_nodes_py/node_with_timer.py
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("node_with_timer")
        self.count_ = 0
        self.timer_ = self.create_timer(0.5, self.timer_callback)
        self.get_logger().info("Node started !")

    def timer_callback(self):
        self.count_ += 1
        self.get_logger().info(f"Hello n°{self.count_}")

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: ros2_fundamentals/start_with_simple_nodes/src/node_with_timer.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("node_with_timer"), count_(0)
    {
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(500),
            std::bind(&MyNode::timerCallback, this)
        );
        RCLCPP_INFO(this->get_logger(), "Node started !");
    }

private:
    void timerCallback()
    {
        count_++;
        RCLCPP_INFO(this->get_logger(), "Hello n:%d", count_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int count_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MyNode>());
    rclcpp::shutdown();
    return 0;
}